cincinnati/src/plugins/catalog.rs

```diff
@@ -0,0 +1,64 @@
+//! Plugins catalog.
+//!
+//! This catalog relies on a static list of all available plugins,
+//! referenced by name. It is used for configuration purposes.
+
+use super::internal::channel_filter::ChannelFilterPlugin;
+use super::internal::edge_add_remove::EdgeAddRemovePlugin;
+use super::internal::metadata_fetch_quay::QuayMetadataFetchPlugin;
+use super::internal::node_remove::NodeRemovePlugin;
+use super::{Plugin, PluginIO};
+use failure::Fallible;
+use std::fmt::Debug;
+
+/// Key used to look up plugin-type in a configuration entry.
+static CONFIG_PLUGIN_NAME_KEY: &str = "name";
+
+/// Settings for a plugin.
+pub trait PluginSettings: Debug + Send {
+    fn build_plugin(&self) -> Fallible<Box<Plugin<PluginIO>>>;
+}
```

Comment:
Could we add the `deserialize_config` function and the `PLUGIN_NAME` variable as part of the trait?

Reply:
Negative answer for both of them, because:
```
the trait `plugins::catalog::PluginSettings` cannot be made into an object
   |
   = note: method `deserialize_config` has no receiver
   = note: the trait cannot contain associated consts like `PLUGIN_NAME`
```